File: Main/static/JS/game/effects/silence.js
import { Modificator } from "../classes/modificator.js";
import { SummaryDurableEffect } from "./effect.js";


class Silence extends SummaryDurableEffect {
    constructor(hero, game_state, duration, caster){
        super('безмолвие', 'silence.png', hero, game_state, duration, caster)
    }

    gen_description(){
        return `Безмолвие (длительность = ${this.duration}) (наложено: ${this.caster.name})`
    }

    before_move(){
        this.decrease_duration()
    }

    gen_modificators(){
        return [new Modificator('silence', 1, true, 'after', 'set')]
    }
}

export {Silence}

File: Main/models/effects/silence.py
from .effect import SummaryDurableEffect
from django.db import models
from Main.models.modifier import Modifier


class Silence(SummaryDurableEffect):
    name = 'Silence'
    img_src = 'silence.png'

    def gen_description(self, link):
        return f"Безмолвие ({self.duration} ходов) (наложено {link.caster.name})"

    def after_target_move(self, link):
        self.decrease_duration(link)

    def gen_modifiers(self, link):
        return [Modifier(hero=link.hero, param_name='silence', value=True, duration=1, is_set=True)]
